Adds a new brand of cigarettes, Mainthol Greys

massaheartsu opened the pull request:
**We all like a little flavor, and a new brand in the sector promises a dark smoke, questionable trace compounds, and the robust flavors of maintenance...**
![paintdotnet_GC1kzxYiG0](https://github.com/user-attachments/assets/fb1c9986-7ce2-4ab4-a033-f4d2cf93b3c6)

## About The Pull Request
I've added a new brand of cigarettes! Mainthol greys, now available in hacked cigarette vending machines for an immodest sum, hand rolled & grown in the Sybil sector. The syndicate stock some stolen surplus too. The goal was to create another nice little flavor item, a compliment to grey bull and other maintenance loot. 

<details>
  <summary>Chemical contents in code, for the joy of discovery!</summary>

  ```
1u oculine is in every cigarette, an imperceptible buffer against the welder harming an enterprising assistant's eyes. 
Extremely trace amount for cigarette microdosing.
Each cigarette has a 40% chance to have an extra reagent from a list of 14 minor toxins, drugs, and things like glitter. 
They're inconsequential via cigarette, but still noticed and experienced. 
  ```
</details>
I tested these bad boys extensively, blending them, ate them, blinded myself with a welder, spammed a health analyzer while smoking to gauge chemicals and ensure intended effects are noticed and add a nice je ne sais quois.

**Our sprites!**
![image](https://github.com/user-attachments/assets/4a8c8e57-634b-4c29-ae03-ae8913040279)

## Why It's Good For The Game
It's an extra little flavor item with a couple simple surprises to spice up your assisting.
## Changelog
:cl:
add: Added a new brand of cigarettes to hacked & syndicate vending machines+spawner list.
/:cl:

massaheartsu committed:
laying the ground work

added cigarette, cig pack code, added them to spawners, included on vendor lists

massaheartsu committed:
added sprites

added assets needed to test in game

massaheartsu committed:
Update cigarettes.dm

massaheartsu committed:
expanded code

updated chems for final testing

massaheartsu committed:
finalizations

Sprited, scripted, tested, welded & blinded myself, smoked, fent'd, tasted glitter.

Finished sprites, cleaned up ones I wasn't going to use. Finalized chem dosages after significant testing with health analyzers and grinding the ciggies. Cleaned up code. Tested intended and unintended effects. Final draft.

massaheartsu committed:
final draft!

Finalized finalizations for pre-finalization.

Final touches on sprites, ensured everything worked, adjusted a few numbers, descriptions, names.

massaheartsu committed:
finalized pixels

i forgot a sprite

massaheartsu committed:
Update cigarettes.dm

fixed space indent lol

massaheartsu committed:
Update cigarettes.dm

fixed more space indents omg

massaheartsu committed:
Merge branch 'master' into smokes-project

massaheartsu committed:
Update cigarettes.dm

fixed icon state issue

massaheartsu committed:
Merge branch 'smokes-project' of https://github.com/massaheartsu/cozystation into grey-smokes

massaheartsu committed:
another fix lmao

oh my god

massaheartsu committed:
FINAL ADJUSTMENT

Changed name from grey mainthols to mainthol greys.

massaheartsu committed:
post touches

added one new reagent and fixed the crunched box sprite

massaheartsu commented:
I finalized my finished finishing touches for finalizationing.

massaheartsu committed:
Merge branch 'master' into smokes-project

massaheartsu commented:
> looks pretty cool, and random sloppa cigs was something that we have been missing the last sprite looks kinda ugly tho

I made a new one I like more, it IS squished and sideways though! I've only got so many pixels.

sippykot commented:
ok the new sprite is much better

massaheartsu committed:
forgot

forgot to actually add the 14th reagent lol this is actually done now for real

massaheartsu committed:
Merge branch 'smokes-project' of https://github.com/massaheartsu/cozystation into grey-smokes

massaheartsu commented:
Actual final touches, finished what I forgot. All good to go!

timothymtorres commented on code/game/objects/items/cigarettes.dm:
```diff
@@ -637,6 +637,51 @@ CIGARETTE PACKETS ARE IN FANCY.DM
+	list_reagents = null
+
+/obj/item/cigarette/greytide/Initialize(mapload)
+	. = ..()
+	reagents?.add_reagent(/datum/reagent/drug/nicotine, 15)
+	reagents?.add_reagent(/datum/reagent/medicine/oculine, 1) // Very small so the healing is almost imperceptible.
+
+	if (prob(40)) //This rolls a 40% chance to include one of these chems for each cigarette in a pack.
+		var/list/chems = list(
+			/datum/reagent/toxin/fentanyl = 3,
+			/datum/reagent/glitter = 3,
+			/datum/reagent/drug/aranesp = 2,
+			/datum/reagent/consumable/laughter = 2,
+			/datum/reagent/medicine/insulin = 2,
+			/datum/reagent/drug/maint/powder = 2,
+			/datum/reagent/drug/maint/sludge = 2,
+			/datum/reagent/toxin/staminatoxin = 2,
+			/datum/reagent/toxin/leadacetate = 2,
+			/datum/reagent/drug/space_drugs = 2,
+			/datum/reagent/drug/pumpup = 2,
+			/datum/reagent/drug/kronkaine = 2,
+			/datum/reagent/consumable/mintextract = 2,
+			/datum/reagent/pax = 1
+		)
+
+		var/total_weight = 0
+		for (var/chem in chems)
+			total_weight += chems[chem]
+
+		var/random_roll = rand(1, total_weight)
+		var/cumulative = 0
+		var/chosen = null
+
+		for (var/chem in chems)
+			cumulative += chems[chem]
+			if (random_roll <= cumulative)
+				chosen = chem
+				break
+
+		reagents?.add_reagent(chosen, rand(10, 15))
```
```suggestion
/obj/item/cigarette/greytide
	name = "grey mainthol"
	desc = "Made by hand, has a funky smell."
	lung_harm = 2.5
	list_reagents = list(/datum/reagent/drug/nicotine = 15, /datum/reagent/medicine/oculine = 1)
	var/list/static/possible_reagents = list(
		/datum/reagent/toxin/fentanyl = 3,
		/datum/reagent/glitter = 3,
		/datum/reagent/drug/aranesp = 2,
		/datum/reagent/consumable/laughter = 2,
		/datum/reagent/medicine/insulin = 2,
		/datum/reagent/drug/maint/powder = 2,
		/datum/reagent/drug/maint/sludge = 2,
		/datum/reagent/toxin/staminatoxin = 2,
		/datum/reagent/toxin/leadacetate = 2,
		/datum/reagent/drug/space_drugs = 2,
		/datum/reagent/drug/pumpup = 2,
		/datum/reagent/drug/kronkaine = 2,
		/datum/reagent/consumable/mintextract = 2,
		/datum/reagent/pax = 1
	)

/obj/item/cigarette/greytide/Initialize(mapload)
	. = ..()

	if(prob(40))
		reagents.add_reagent(pick_weight(possible_reagents), rand(10, 15))
```

Try this.

massaheartsu committed:
Update code/game/objects/items/cigarettes.dm

Co-authored-by: Tim <timothymtorres@gmail.com>

massaheartsu committed:
Cleaned up the code at timothymtorres' recommendations!

He recommended cleaning up the code structure. I flattened the chances too.

massaheartsu committed:
Merge branch 'smokes-project' of https://github.com/massaheartsu/cozystation into grey-smokes

massaheartsu committed:
Update cigarettes.dm

linted

massaheartsu committed:
Merge branch 'master' into smokes-project

sippykot commented:
also, maybe they should have a bit of menthol in them? because of the name, it sounds like menthol

massaheartsu committed:
ADDED MENTHOL

cries

massaheartsu committed:
Merge branch 'smokes-project' of https://github.com/massaheartsu/cozystation into grey-smokes

massaheartsu commented:
> also, maybe they should have a bit of menthol in them? because of the name, it sounds like menthol

This was an embarrassing oversight! I have corrected it post haste! Thank u.

CCC23 commented:
smoking that pack that got axel brady qced I have seen the enhanced interogation room I beat the case this shit aint nothing to me man

carlarctg commented:
not sure about the neon green line on the pack, not very mainty. why not yellow to look like a maint airlock?

massaheartsu commented:
> not sure about the neon green line on the pack, not very mainty. why not yellow to look like a maint airlock?

which neon green line

carlarctg commented:
the thingy

massaheartsu commented:
> the thingy

I wish I knew what you meant! There's only a thin grey line, otherwise any green is the eye plates of the gas mask on the front of the carton.

Loafin34 commented:
The sprites look really crisp and the cigarette idea is in-line with what tiders expect! I hope the grey gets merged soon!

MrMelbert commented on code/game/objects/effects/spawners/random/entertainment.dm:
```diff
@@ -151,6 +151,7 @@
 		/obj/item/storage/fancy/cigarettes/cigpack_robustgold = 1,
 		/obj/item/storage/fancy/cigarettes/cigpack_midori = 1,
 		/obj/item/storage/fancy/cigarettes/cigpack_candy = 1,
+		/obj/item/storage/fancy/cigarettes/cigpack_greytide = 1
```
```suggestion
		/obj/item/storage/fancy/cigarettes/cigpack_greytide = 1,
```

MrMelbert commented on code/game/objects/effects/spawners/random/entertainment.dm:
```diff
@@ -165,6 +166,7 @@
 		/obj/item/cigarette/robust = 2,
 		/obj/item/cigarette/carp = 1,
 		/obj/item/cigarette/robustgold = 1,
+		/obj/item/cigarette/greytide = 3
```
```suggestion
		/obj/item/cigarette/greytide = 3,
```

MrMelbert commented on code/game/objects/items/cigarettes.dm:
```diff
@@ -637,6 +637,32 @@ CIGARETTE PACKETS ARE IN FANCY.DM
+		/datum/reagent/consumable/mintextract = 2,
+		/datum/reagent/pax = 1
+	)
+/obj/item/cigarette/greytide/Initialize(mapload)
```
```suggestion

/obj/item/cigarette/greytide/Initialize(mapload)
```

MrMelbert commented on code/modules/vending/cigarette.dm:
```diff
@@ -19,6 +19,7 @@
 	contraband = list(
 		/obj/item/vape = 5,
 		/obj/item/cigarette/dart = 1,
+		/obj/item/storage/fancy/cigarettes/cigpack_greytide = 1
```
```suggestion
		/obj/item/storage/fancy/cigarettes/cigpack_greytide = 1,
```

MrMelbert commented on code/modules/vending/cigarette.dm:
```diff
@@ -43,9 +44,10 @@
 		/obj/item/storage/box/matches = 10,
 		/obj/item/lighter/greyscale = 4,
-		/obj/item/storage/fancy/rollingpapers = 5,
+		/obj/item/storage/fancy/rollingpapers = 5
```
```suggestion
		/obj/item/storage/fancy/rollingpapers = 5,
```

MrMelbert commented on code/game/objects/items/cigarettes.dm:
```diff
@@ -637,6 +637,32 @@ CIGARETTE PACKETS ARE IN FANCY.DM
+	desc = "Made by hand, has a funky smell."
+	lung_harm = 2.5
+	list_reagents = list(/datum/reagent/drug/nicotine = 15, /datum/reagent/consumable/menthol = 6, /datum/reagent/medicine/oculine = 1)
+	var/static/list/possible_reagents = list(
```
Document this
Something like `/// Weighted list of random reagents to add`

massaheartsu replied:
Thank you! I got these changes applied.

massaheartsu committed:
Update code/game/objects/effects/spawners/random/entertainment.dm

Co-authored-by: MrMelbert <51863163+MrMelbert@users.noreply.github.com>

massaheartsu committed:
Update code/modules/vending/cigarette.dm

Co-authored-by: MrMelbert <51863163+MrMelbert@users.noreply.github.com>

massaheartsu committed:
Update code/modules/vending/cigarette.dm

Co-authored-by: MrMelbert <51863163+MrMelbert@users.noreply.github.com>

massaheartsu committed:
Update code/game/objects/effects/spawners/random/entertainment.dm

Co-authored-by: MrMelbert <51863163+MrMelbert@users.noreply.github.com>

massaheartsu committed:
Update code/game/objects/items/cigarettes.dm

Co-authored-by: MrMelbert <51863163+MrMelbert@users.noreply.github.com>

massaheartsu committed:
The third finalized finishizations of finalimentality

Commited Melbert's recommended changes, added chem volume limit just to ensure there's enough space for the chems, added documentation to list of ciggy changes.

massaheartsu committed:
Merge branch 'master' into smokes-project

massaheartsu committed:
Merge branch 'master' into smokes-project

san7890 reviewed (commented):
most minor nit ever

massaheartsu committed:
Update code/game/objects/items/cigarettes.dm

Co-authored-by: san7890 <the@san7890.com>
